src/Compilers/Core/Portable/Syntax/InternalSyntax/ISyntaxList.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Syntax.InternalSyntax
+{
+    /// <summary>
+    /// Interface that allows us to operate generically over VB or C# SyntaxLists.
+    /// </summary>
+    internal interface ISyntaxList<TGreenNode>
+        where TGreenNode : GreenNode
```

Comment:
Introducing interface dispatch at this level typically results in perf regressions. Efforts were made to not have interfaces here. 
Is it possible to not bring interfaces back?